src/components/RealTimeInterface/TimePicker.vue

```diff
@@ -1,33 +1,111 @@
 <script setup>
-import { ref, watch, defineEmits } from 'vue'
+import { ref, watch, defineEmits, onMounted, computed } from 'vue'
 import { useRouter } from 'vue-router'
 import { useSessionsStore } from '../../stores/sessions'
 import { fetchApiCall } from '../../utils/api'
-import LeafletMap from './GlobeMap/LeafletMap.vue'
-import { formatToUTC, formatDate } from '../../utils/formatTime.js'
+import { formatToUTC, formatDate, formatTime } from '../../utils/formatTime.js'
 import { useConfigurationStore } from '../../stores/configuration'
+import LeafletMap from './GlobeMap/LeafletMap.vue'
 
 const router = useRouter()
 const sessionsStore = useSessionsStore()
 const configurationStore = useConfigurationStore()
 
 const date = ref(null)
 const startTime = ref(null)
-const endDate = ref(null)
 const errorMessage = ref(null)
 const selectedSite = ref(null)
-const emits = defineEmits(['changeView'])
-
-// TO DO: Get times from API
-const times = ['10:45', '11:00', '11:15', '11:30', '11:45', '12:00', '12:15', '12:30', '12:45', '13:00', '13:15', '13:30', '13:45', '14:00', '14:15', '14:30', '14:45', '15:00', '15:15', '15:30', '15:45', '16:00', '16:15', '16:30', '16:45', '17:00', '17:15', '17:30', '17:45', '18:00', '18:15', '18:30', '18:45', '19:00', '19:15', '19:30']
-
-const setEndDate = (startDate, startTime, minutesToAdd) => {
-  const combinedDateTime = new Date(startDate)
-  const [hours, minutes] = startTime.split(':')
-  combinedDateTime.setHours(hours)
-  combinedDateTime.setMinutes(minutes)
-  combinedDateTime.setMinutes(combinedDateTime.getMinutes() + minutesToAdd)
-  return combinedDateTime.toISOString().split('.')[0] + 'Z'
+const availableTimes = ref({})
+const localTimes = ref([])
+
+const emits = defineEmits(['timeSelected'])
+
+const hasAvailableTimes = computed(() => {
+  return Object.keys(availableTimes.value).length > 0
+})
+
+// Rounds times from ocs to the nearest 15 minutes
+function roundToNearest15Minutes (date, direction = 'up') {
+  // 15 minutes in milliseconds
+  const ms = 1000 * 60 * 15
+  const roundedDate = new Date(Math[direction === 'up' ? 'ceil' : 'floor'](date.getTime() / ms) * ms)
+  return roundedDate
+}
+
+// Because we get availability in ranges, we need to generate all the 15 minute intervals within those ranges
+function generate15MinuteIntervals (startDate, endDate) {
+  const intervals = []
+  let current = new Date(startDate)
+  while (current <= endDate) {
+    intervals.push(current.toISOString())
```

Comment:
Why not just store the javascript Date object internally, and just convert to a string when you want to display it or send it to an API? I say this because by storing it as a string in this internal object, you end up having to turn it back into a Date object later on to get the locale date both here and in the LeafletMap.

Reply:
solid point